### ISSUE TITLE
Rename References in GlassBR

### ISSUE THREAD

**Issue by samm82** [JacquesCarette/Drasil#721]
Similar to smiths/caseStudies#104, the ASTM references should potentially be updated in Drasil, although they are better than the ones in caseStudies were. The main problems are the author field (should be "ASTM") and the name of them (should be "ASTMyear").

https://github.com/JacquesCarette/Drasil/blob/bda79c7dae9c845c178e0c309dca68ec0f4bec4a/code/Example/Drasil/GlassBR/References.hs#L19-L43

@JacquesCarette @smiths Should I proceed with this? If so, this should be done before smiths/caseStudies#31.

**Comment by smiths** [comment]
@samm82, I think you should proceed with this.  I can't see any reason why the new bibliographic information isn't better than the old information.

**Comment by samm82** [comment]
@smiths Should I also rename this:

https://github.com/JacquesCarette/Drasil/blob/bda79c7dae9c845c178e0c309dca68ec0f4bec4a/code/Example/Drasil/GlassBR/References.hs#L45-L52

to `wlBeasonEtAl1998`, or is there some other convention you'd like me to use?

**Comment by smiths** [comment]
Yes, please rename the citation.  I find it much easier to read citation names when a standard convention is followed.

**Comment by samm82** [comment]
@smiths The rename of `glThick1998` to `wlBeasonEtAl1998` (Beason, W. Lynn) causes the references to no longer be in alphabetical order:

## Stable
**[4]: Beason, W. Lynn, Kohutek, Terry L., and Bracci, Joseph M. Basis for ASTME E 1300 Annealed Glass Thickness Selection Charts. ASCE Library. February, 1998. doi.org/10.1061/(ASCE)0733-9445(1998)124:2(215).**
[5]: Koothoor, Nirmitha. A document drive approach to certifying scientific computing software. McMaster University, Hamilton, ON, Canada: 2013. Print.
[6]: Parnas, David L. and Clements, P. C. "A rational design process: How and why to fake it." IEEE Transactions on Software Engineering, vol. 12, no. 2, Washington, USA: February, 1986. pp. 251–257. Print.
[7]: Robertson, James and Robertson, Suzanne. Volere requirements specification template edition 16. 2012. https://pdfs.semanticscholar.org/cf57/27a59801086cbd3d14e587e09880561dbe22.pdf.
[8]: Smith, W. Spencer and Lai, Lei. "A new requirements template for scientific computing." Proceedings of the First International Workshop on Situational Requirements Engineering Processes - Methods, Techniques and Tools to Support Situation-Specific Requirements Engineering Processes, SREP'05. Edited by PJ Agerfalk, N. Kraiem, and J. Ralyte, Paris, France: 2005. pp. 107–121. In conjunction with 13th IEEE International Requirements Engineering Conference,

## Build
[4]: Koothoor, Nirmitha. A document drive approach to certifying scientific computing software. McMaster University, Hamilton, ON, Canada: 2013. Print.
[5]: Parnas, David L. and Clements, P. C. "A rational design process: How and why to fake it." IEEE Transactions on Software Engineering, vol. 12, no. 2, Washington, USA: February, 1986. pp. 251–257. Print.
[6]: Robertson, James and Robertson, Suzanne. Volere requirements specification template edition 16. 2012. https://pdfs.semanticscholar.org/cf57/27a59801086cbd3d14e587e09880561dbe22.pdf.
[7]: Smith, W. Spencer and Lai, Lei. "A new requirements template for scientific computing." Proceedings of the First International Workshop on Situational Requirements Engineering Processes - Methods, Techniques and Tools to Support Situation-Specific Requirements Engineering Processes, SREP'05. Edited by PJ Agerfalk, N. Kraiem, and J. Ralyte, Paris, France: 2005. pp. 107–121. In conjunction with 13th IEEE International Requirements Engineering Conference,
**[8]: Beason, W. Lynn, Kohutek, Terry L., and Bracci, Joseph M. Basis for ASTME E 1300 Annealed Glass Thickness Selection Charts. ASCE Library. February, 1998. doi.org/10.1061/(ASCE)0733-9445(1998)124:2(215).**

Is this an issue?

**Comment by smiths** [comment]
What determines the order of the references in the bibliography in the build version?  With normal bibtex style files the order is a variability that the user can control.  The order could be in the order of usage in the paper, or a sorted list.  I don't think Drasil has a variability for the order, but some decision should be made.  Sorted alphabetically come up most frequently.  If the Drasil (build) order is just random, I would prefer that they be sorted alphabetically.  Is that feasible?

**Comment by JacquesCarette** [comment]
This might be an oddity of the HTML printer, as opposed to the LaTeX -- am I right?

**Comment by samm82** [comment]
No - it occurs in the LaTeX too.

**Comment by samm82** [comment]
I'm assuming that the names of the references get alphabetized, so changing `glThick1998` to `wlBeasonEtAl1998` kicked it to the bottom of the list, when they should be alphabetized by author name - not sure how to do this/where this happens though.

**Comment by JacquesCarette** [comment]
That's weird.  I suspect that something is odd with the wlBeason author that causes this glitch.

**Comment by samm82** [comment]
To stick with the consistent format, it should have been renamed to `beasonEtAl1998` instead of `wlBeasonEtAl1998`, which fixes the issue.

**Comment by JacquesCarette** [comment]
The alphabetization should definitely be by author name and not by reference id.  Look for uses of `sortBy` in the code. There are rather few left.

**Comment by JacquesCarette** [comment]
Nice hack.  But we should figure out how to fix the deeper issue.

**Comment by samm82** [comment]
Is there a deeper issue though? The references should be sorted by last name, and the references themselves should prefaced by the last name, so there should never be this conflict. However, I agree that we should explicitly sort by author.

**Comment by JacquesCarette** [comment]
If changing `wlBeasonEtAl1998` to `beasonEtAl1998` fixed it, then yes, definitely. We shouldn't sort by `uid`!

**Comment by samm82** [comment]
Should this be a separate issue? I just pushed a change that addresses this issue. Should I just keep working on this issue on the same branch?

**Comment by samm82** [comment]
I'm also planning to work on smiths/caseStudies#31 on the GlassReferences branch as well - is that a good idea?

**Comment by JacquesCarette** [comment]
It is a separate issue, and should have been solved independently, so that it can be merged sooner rather than later. Just in case something happens and the branch that you are currently working on doesn't get merge for a long time.  It is best to work on independent fixes in independent branches.

**Comment by samm82** [comment]
Closed with merge of #728

**Comment by smiths** [comment]
I agree that this issue can now be closed.

**Comment by samm82** [comment]
@smiths Can I merge the branch then? Or should I wait for @JacquesCarette to?

**Comment by smiths** [comment]
Oops.  I thought it was already merged.  I cannot see any reason why @JacquesCarette won't do the merge, but we better respect the process and give him a chance to do the merge.

**Comment by JacquesCarette** [comment]
Done.
